daml-lf/parser/src/main/scala/com/digitalasset/daml/lf/testing/parser/ExprParser.scala

```diff
@@ -149,14 +149,13 @@ private[parser] class ExprParser[P](parserParameters: ParserParameters[P]) {
     }
 
   private lazy val eVariantOrEnumCon: Parser[Expr] =
-    fullIdentifier ~ (`:` ~> id) ~ rep(argTyp) ~ opt(expr0) ^^ {
-      case tName ~ vName ~ argsTyp ~ Some(arg) =>
-        EVariantCon(TypeConApp(tName, argsTyp.to(ImmArray)), vName, arg)
-      case _ ~ _ ~ argsTyp ~ None if argsTyp.nonEmpty =>
-        throw new java.lang.Error("enum type do not take type parameters")
-      case tName ~ vName ~ _ ~ None =>
-        EEnumCon(tName, vName)
-    }
+    fullIdentifier ~ (`:` ~> id) ~ rep1(argTyp) ~> err("enum type does not take type parameters") |
+      fullIdentifier ~ (`:` ~> id) ~ rep(argTyp) ~ opt(expr0) ^^ {
+        case tName ~ vName ~ argsTyp ~ Some(arg) =>
```

Comment:
You will reject valid variant.

Reply:
Good point. We need to capture end of input in order for this rule to be valid.

Reply:
Used `phrase` to ensure whole input consumption before triggering an error